frontend/views/utils/validators.js

```diff
@@ -1,3 +1,19 @@
+// Only allows non-empty strings of plain ASCII letters and digits.
+export const alphanumeric = (value: string): boolean => /^[A-Za-z\d]+$/.test(value)
+
+// Same as `alphanumeric`, but also allows accented letters from the first 256 Unicode code points.
+export const alphanumericLatin = (value: string): boolean => /^[A-Za-zÀ-ÖØ-Ýà-öø-ÿ\d]+$/.test(value)
+
+// Same as `alphanumeric`, but also allows hyphens.
+export const alphanumericOrHyphens = (value: string): boolean => /^[A-Za-z\d-]+$/.test(value)
```

Comment:
What about underscores? Can we have those? (I personally see no issue with them)

Reply:
Sure, but the name of this function is going to become a bit long. Any idea to avoid `alphanumericOrHyphensOrUnderscores` ?

Reply:
To simplify things, maybe remove `alphanumericLatin` and then rename `alphanumericOrHyphens` to something more generic and simple like `isValidUsername`

Reply:
There are more conditions for a string to actually be a valid username, so that name wouldn't really be warranted. Maybe `allowedUsernameCharacters`?

Reply:
Yeah that sounds good 👍

Reply:
Also, should we apply the same rules to underscores as for hyphens, i.e. no leading, trailing, or consecutive underscores?

Reply:
Sure. And we can relax or modify these rules in the future if necessary.

Reply:
Yeah. It's likely better to start strict and relax as necessary than the opposite, as otherwise previously valid usernames could become suddenly invalid.